Initial implementation

gohabereg opened the pull request.

gohabereg committed:
Initial implementation

neSpecc committed:
Some updates

- Inline Toolbar supported
- Sup style updated
- Popover style updated
- Icon added

talyguryn commented on package.json:
```diff
@@ -0,0 +1,36 @@
+    "webpack-cli": "^4.6.0"
+  },
+  "dependencies": {
+    "@codexteam/shortcuts": "^1.1.1"
```
need to move this to devdeps too, as i understand. because you are going to use bundle not sources

gohabereg replied:
It is included to bundle, so should be in deps

talyguryn committed:
Create npm-publish.yml

gohabereg committed:
Add shortcut

gohabereg committed:
Add eslint workflow

gohabereg committed:
Merge branch 'initial-implementation' of github.com:editor-js/footnotes-tune into initial-implementation

gohabereg committed:
Add integration tests via cypress

gohabereg committed:
Update editorjs package to remote

gohabereg commented:
Tests will not work on CI until core branch is merged to develop and published to npm 😢

gohabereg committed:
Fix bugs with editor events

gohabereg committed:
Improvements

- Add Apply & Remove button
- Apply change on CMD+Enter
- Add external config support
- Better positioning & popover movement on resize
- Add data attribute for sup elements
- Hide popover on second click on note

gohabereg committed:
Add more tests & coverage

neSpecc commented on README.md:
````diff
@@ -1,2 +1,129 @@
+```
+
+### Load from CDN
+Load the script from [jsDelivr](https://www.jsdelivr.com/package/npm/@editorjs/footnote) CDN and connect to your page.
````
```suggestion
Load the script from [jsDelivr](https://www.jsdelivr.com/package/npm/@editorjs/footnotes) CDN and connect to your page.
```

neSpecc commented on README.md:
```diff
@@ -1,2 +1,129 @@
+### Load from CDN
+Load the script from [jsDelivr](https://www.jsdelivr.com/package/npm/@editorjs/footnote) CDN and connect to your page.
+
+<script src="https://cdn.jsdelivr.net/npm/@editorjs/footnote@latest"></script>
```
```suggestion
<script src="https://cdn.jsdelivr.net/npm/@editorjs/footnotes@latest"></script>
```

neSpecc commented on README.md:
```diff
@@ -1,2 +1,129 @@
+    "text": "You text with a footnote<sup data-tune=\"footnote\">1</sup>. And another footnote<sup data-tune=\"footnote\">2</sup>"
+  },
+  "tunes": {
+    "footnote": [
```
Footnote or Footnotes? Maybe we need to choose one version and rename other places?

neSpecc commented on package.json:
```diff
@@ -3,21 +3,25 @@
+    "@cypress/code-coverage": "^3.9.9",
     "@cypress/webpack-preprocessor": "^5.9.1",
-    "@editorjs/editorjs": "../editor.js-copy",
+    "@editorjs/editorjs": "https://github.com/codex-team/editor.js#feature/tunes-improvements",
```
do not forget to remove branch before merge

gohabereg committed:
Footnote -> footnotes, fix popover positioning

gohabereg committed:
Update styles and enter behaviour

gohabereg committed:
Update z-index

gohabereg committed:
Add confirm before note removal

gohabereg committed:
Remove console log

gohabereg committed:
Add fake cursor animation from core & use editor package from npm

gohabereg committed:
Adopt some tests for CI

gohabereg committed:
Skip some tests for FF

gohabereg committed:
Update readme with note on FF
